src/index.js

```diff
@@ -1,92 +1,172 @@
+  }
+
+  return {
+    graphState: _graphState,
+    initGraph: initGraph,
+    updateState: updateState
+  }
+})();
+
+// remove and replace once graph state function fully developed
+GraphStateCtrl.initGraph(0);
+
+//// RESPONSIBLE FOR WATCHING AND RENDERING ELEMENTS IN THE UI
+const UICtrl = (function() {
+  const contributionGraph = document.getElementById("year");
+  const textForm = document.getElementById("text-form");
+  const textField = document.getElementById("text-field");
 
-        // INCREASE COUNT OF YEAR FOR ID-ING DAYS
-        dayOfYear++;
-      } else {
+  const graphClearButton = document.getElementById("clear-button");
+  const logGraphStateButton = document.getElementById("log-graphState");
+  const drawRadioOption = document.getElementById("draw-option");
+  const eraseRadioOption = document.getElementById("erase-option");
+  const fullRadioOption = document.getElementById("full-option");
+
+  // TO DO:
+  // (function() {clear/init empty schedule})();
+  // function renderSchedule (Schedule) {render schedule}
+  // userNameInput.addeventlistener('submit', function(e){console.log(e.target.value); e.target.value = '';})
+  // scheduleRequest.addeventlistener('submit', renderSchedule(Schedule))
+
+  window.onload = () => {
+    drawGraph()
+  }
+
+  let drawGraph = () => {
+    // deletes graph if already present
+    if(contributionGraph.hasChildNodes()) {
+      while(contributionGraph.firstChild) {
+        contributionGraph.firstChild.remove();
       }
     }
-    // APPEND WEEK TO GRAPH
-    contributionGraph.appendChild(createdWeek);
+    // draws graph in UI based off current graphState
+    for(let Week in GraphStateCtrl.graphState) {
+      let currentlyEvaluatedWeek = GraphStateCtrl.graphState[Week];
+      const createdWeek = document.createElement("div");
+      createdWeek.id = Week;
+      let day = 0;
+      currentlyEvaluatedWeek.forEach(dayValue => {
+        const createdDay = document.createElement("div");
+        createdDay.id = day;
+        createdDay.className = `commit-${dayValue}`;
+        createdWeek.appendChild(createdDay);
+        day++
+      });
+      contributionGraph.appendChild(createdWeek);
+    };
+  }
+
+  let graphClear = (event) => {
+    event.preventDefault()
+    GraphStateCtrl.initGraph(0);
+    drawGraph();
+  }
+
+  // update to include change in color over time held
+  let drawSwitch = true;
+  let eraseSwitch = false;
+  let fullSwitch = false;
+  let drawOptionSwitcher = (event) => {
+    let optionToTurnOn = event.target.id.replace('-option', 'Switch');
+    if(optionToTurnOn === "eraseSwitch") {
+      eraseSwitch = true;
+      fullSwitch = false;
+      drawSwitch = false;
+    } else if(optionToTurnOn === "fullSwitch") {
+      fullSwitch = true;
+      eraseSwitch = false;
+      drawSwitch = false;
+    } else if(optionToTurnOn === "drawSwitch") {
+      drawSwitch = true;
+      eraseSwitch = false;
+      fullSwitch = false;
+    }
+  }
```

Comment:
2 valuable structural lessons here; (the latter definitely more applicable then the other, but both are interesting)

`let optionToTurnOn = event.target.id.replace('-option', 'Switch');` is definitely hard to read and comes of a little bit hacky.

I notice that the `drawOptionSwitcher` function is used only for the following event listeners;

```
drawRadioOption.addEventListener("mouseup", drawOptionSwitcher);  eraseRadioOption.addEventListener("mouseup", drawOptionSwitcher);  fullRadioOption.addEventListener("mouseup", drawOptionSwitcher);
```

One thing you should comfortable at is inlining a little more info at the function callers side. Function caller here being the event listener.

Like
```
drawRadioOption.addEventListener("mouseup", () => drawOptionSwitcher("drawSwitch"));
eraseRadioOption.addEventListener("mouseup",() => drawOptionSwitcher("eraseSwitch");  fullRadioOption.addEventListener("mouseup", () => drawOptionSwitcher("fullSwitch");
```

---

Your switch approach can be improved. But I know as potentially further switches like this get added, it's going to start looking like a Boeing 747 cockpit in here real quick. Typically it's fine when all switches work independently, but in your case, it doesn't make sense to have `drawSwitch` and `eraseSwitch` both turn on for example.

Check out the idea of having a `drawMode` that can be either in _standard_, _erase_, _bold_ mode. You can implement this with clearly with some constants and then a string variable like this;

```
const NORMAL, ERASE, BOLD = "normal", "erase", "bold" // all caps informs reader of never changing constants (follows normal convention)
...
if drawMode === STANDARD {
   ...
}
...
if (isDrawing && drawMode === STANDARD) {
   ...
}
...
drawRadioOption.addEventListener("mouseup", () => updateDrawMode(STANDARD));
eraseRadioOption.addEventListener("mouseup",() => updateDrawMode(ERASE);  fullRadioOption.addEventListener("mouseup", () => updateDrawMode(BOLD);
```